dbt/include/athena/macros/materializations/models/table/table.sql

```diff
@@ -133,20 +133,27 @@
         {%- endif -%}
 
         {%- set old_relation_table_type = adapter.get_glue_table_type(old_relation) -%}
+        -- we cannot use old_bkp_relation, because it returns None if the relation doesn't exist
+        -- we need to create a python object via the make_temp_relation instead
+        {%- set old_relation_bkp = make_temp_relation(old_relation, '__bkp') -%}
 
-        {%- if old_relation_table_type == 'iceberg' -%}
-          {{ rename_relation(old_relation, old_bkp_relation) }}
+        {%- if old_relation_table_type.value == 'iceberg_table' -%}
+          {{ rename_relation(old_relation, old_relation_bkp) }}
         {%- else  -%}
           {%- do drop_relation_glue(old_relation) -%}
         {%- endif -%}
 
         {{ rename_relation(tmp_relation, target_relation) }}
 
-        -- old_bkp_relation might not exists in case we have a switch from hive to iceberg
-        -- we prevent to drop something that doesn't exist even if drop_relation is able to deal with not existing tables
-        {%- if old_bkp_relation is not none -%}
-          {%- do drop_relation(old_bkp_relation) -%}
+        -- old_relation_bkp might not exists in case we have a switch from hive to iceberg
+        -- we cannot use old_bkp_relation, because result could be cached by dbt, we use instead glue apis
+        -- get_glue_table returns none in case EntityNotFoundException, therefore this works well with restrictive environment
+        -- where Lakeformation is used, and allow operations in only existing objects
+        {%- set old_relation_bkp_exists = adapter.get_glue_table(old_relation_bkp) -%}
```

Comment:
We will get a 403 permission denied code here, as the table doesnt exist, which is different to the EntityNotFoundException. 😭

Reply:
we have 2 options: 
1. we modify get_glue_table to tackle the exception that you have, and in that case return None
2. we create a does_table_exists, that is a wrapper on top of get_glue_table, and it returns True only if the table exists, otherwise it fallback to False, and should cover your edge case.

Reply:
I addressed this comment, please have a look.